### PR TITLE
RNA: Add ThemeProvider component

### PR DESCRIPTION
**Pull request by renatoagds**
Implements the `ThemeProvider` component for Design Tokens variables setup.

#### Changes proposed in this Pull Request:

* Add `ThemeProvider` component

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Go to `Storybook` folder and run it
* Open `Theme Provider` story.

#### Demo

https://user-images.githubusercontent.com/1663717/157136573-5e876d18-6830-4b6b-9237-413b8322c8be.mp4

### CODE REVIEW

**Comment by renatoagds** [comment]
Closing in favor of #22496

**Comment by renatoagds** [comment]
Reopening since we're not going to implement it in #22496

**Comment by oskosk** [comment]
@retrofox @leogermani let's all review this one and be aware of how to use it, and the benefits it will provide

**Review by leogermani — DISMISSED** [review]
This looks great! And having the story to view it all is awesome.

One thing I think is a must have though is a README with a simple code example on how to consume and use it :) (The sory code is a little hard to parse because of all the abstraction to build the loops)

**Comment by renatoagds** [comment]
@leogermani Good point. I'll do that in a follow-up PR, I already have the implementation on My Jetpack that will arrive in the next PR.

**Review by leogermani — APPROVED** [review]
